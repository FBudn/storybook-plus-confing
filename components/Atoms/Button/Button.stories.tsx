import React from "react";
import { View } from "react-native";
import Button from "./Button";

const ButtonMeta = {
  title: "Button",
  component: Button,
  argTypes: {
    onPress: { action: "pressed the button" },
  },
  args: {
    children: "Hello IM HERE love u much",
    testId: "test-button-id",
  },
};

export default ButtonMeta;

export const Basic = {};

export const AnotherExample = {
  args: {
    children: "Another example",
  },
};
