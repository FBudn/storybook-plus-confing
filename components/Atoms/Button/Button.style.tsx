import { StyleSheet } from "react-native";

const ButtonStyles = StyleSheet.create({
  container: {
    maxHeight: 40,
    height: 40,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    backgroundColor: "rgb(236 72 153)",
    shadowColor: "#94a3b8",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.23,
    shadowRadius: 2.62,
    borderRadius: 5,
  },
  text: { color: "white", fontWeight: "bold" },
});

export default ButtonStyles;
