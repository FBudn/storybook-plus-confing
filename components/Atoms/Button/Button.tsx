import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";

export interface ButtonProps {
  onPress: () => void;
  children: string;
  testId: string;
}

function Button({ onPress, children, testId }: ButtonProps) {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      testID={testId}
    >
      <Text style={styles.text}>{children}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 0.1,
    maxHeight: 40,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    backgroundColor: "rgb(236 72 153)",
    shadowColor: "#94a3b8",
    borderRadius: 5,
  },
  text: { color: "white", fontWeight: "bold" },
});

export default Button;
