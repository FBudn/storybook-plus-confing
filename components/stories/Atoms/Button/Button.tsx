/* eslint-disable @typescript-eslint/no-use-before-define */
/* eslint-disable import/prefer-default-export */
import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";

export interface MyButtonProps {
  onPress: () => void;
  text: string;
}

function MyButton({ onPress, text }: MyButtonProps) {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      testID="test-myButton-id"
    >
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: "violet",
  },
  text: { color: "black" },
});

export default MyButton;
