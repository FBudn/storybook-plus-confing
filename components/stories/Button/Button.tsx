/* eslint-disable @typescript-eslint/no-use-before-define */
/* eslint-disable import/prefer-default-export */
import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";

export interface MyButtonProps {
  onPress: () => void;
  text: string;
}

function MyButton({ onPress, text }: MyButtonProps) {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      testID="test-myButton-id"
    >
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: "violet",
  },
  text: { color: "black" },
});

export default MyButton;

/*
import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";

const function MyButton = ({ onPress, text }) => {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      activeOpacity={0.8}
    >
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: "purple",
    borderRadius: 8,
  },
  text: { color: "white" },
});

export default MyButton;
*/
