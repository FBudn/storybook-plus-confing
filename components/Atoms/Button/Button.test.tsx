import React from "react";
import { fireEvent, render, screen } from "@testing-library/react-native";
import "@testing-library/jest-native/extend-expect";
import TestRenderer from "react-test-renderer";
import Button, { ButtonProps } from "./Button";

const mockOnButtonClick = jest.fn();
const props: ButtonProps = {
  onPress: mockOnButtonClick,
  children: `test test`,
  testId: `test-button-id`,
};

test(`should Button render`, () => {
  render(<Button {...props} />);
  const ButtonElement = screen.getByTestId(`test-button-id`);
  expect(ButtonElement).toBeOnTheScreen();
});

test("Should match snapshot:", () => {
  const ButtonElement = TestRenderer.create(<Button {...props} />).toJSON();
  expect(ButtonElement).toMatchSnapshot();
});

test(`Should have styles and render`, () => {
  render(<Button {...props} />);
  const ButtonElement = screen.getByTestId(`test-button-id`);
  expect(ButtonElement).toBeOnTheScreen();
  expect(ButtonElement).toHaveStyle({
    flex: 0.1,
    maxHeight: 40,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    backgroundColor: "rgb(236 72 153)",
    shadowColor: "#94a3b8",
    borderRadius: 5,
  });
});

test(`should render with text `, () => {
  render(<Button {...props} />);
  const ButtonElement = screen.getByTestId(`test-button-id`);
  expect(ButtonElement).toBeOnTheScreen();
  expect(ButtonElement).toHaveTextContent(`test test`);
});

test(`Should render button and handle onClick`, () => {
  render(<Button {...props} />);
  const ButtonElement = screen.getByTestId(`test-button-id`);
  expect(ButtonElement).toBeOnTheScreen();
  expect(mockOnButtonClick).toBeCalledTimes(0);
  fireEvent.press(ButtonElement);
  expect(mockOnButtonClick).toBeCalledTimes(1);
});

// import "@testing-library/jest-native/extend-expect";
