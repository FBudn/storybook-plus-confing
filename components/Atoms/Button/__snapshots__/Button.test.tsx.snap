// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot: 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(236 72 153)",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "margin": 5,
      "maxHeight": 40,
      "opacity": 1,
      "shadowColor": "#94a3b8",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
      "textAlign": "center",
    }
  }
  testID="test-button-id"
>
  <Text
    style={
      {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  >
    test test
  </Text>
</View>
`;
